sign.js

```diff
@@ -67,7 +67,11 @@ module.exports = function (payload, secretOrPrivateKey, options, callback) {
   }
 
   if (!secretOrPrivateKey) {
```

Comment:
What if we modify the if statement to take the that situation into account? Something like `(!secretOrPrivateKey && options.algorithm !== 'none')` what do you think?